realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -441,33 +441,45 @@ private void emitValidateTableMethod(JavaWriter writer) throws IOException {
 
                 // make sure that nullability matches
                 if (metadata.isNullable(field)) {
-                    writer.beginControlFlow("if (!table.isColumnNullable(%s))", fieldIndexVariableReference(field));
-                    if (Utils.isBoxedType(fieldTypeCanonicalName)) {
+                    // Check if the existing PrimaryKey does support null value for String, Byte, Short, Integer, & Long
+                    if (field.equals(metadata.getPrimaryKey()) && Utils.isNullablePrimaryKeyType(field)) {
```

Comment:
There only remains `field.equals(metadata.getPrimaryKey())` to produce a different message for importing non-nullable `PrimaryKey` column from existing a Realm.